waspc/cli/src/Wasp/Cli/Command/CreateNewProject/StarterTemplates.hs

```diff
@@ -7,78 +7,190 @@ module Wasp.Cli.Command.CreateNewProject.StarterTemplates
+    ( \projectDirName ->
+        unlines
+          [ styleText $ "To run your new app, do:",
+            styleCode $ "    cd " <> projectDirName FP.</> "app",
+            styleCode $ "    wasp db start",
+            styleText $ "Then open new terminal window/tab in that same dir and do:",
+            styleCode $ "    wasp db migrate-dev",
+            styleCode $ "    cp .env.server.example .env.server",
+            styleCode $ "    wasp start",
+            styleText $ "",
+            styleText $ "Check the README for additional guidance and link to docs!"
+          ]
+    )
+
+{- HLINT ignore todoTsStarterTemplate "Redundant $" -}
+
+todoTsStarterTemplate :: StarterTemplate
+todoTsStarterTemplate =
+  simpleGhRepoTemplate
+    ("starters", [reldir|todo-ts|])
+    ( "todo-ts",
+      "Simple but well-rounded Wasp app implemented with Typescript & full-stack type safety."
+    )
+    ( \projectDirName ->
+        unlines
+          [ styleText $ "To run your new app, do:",
+            styleCode $ "    cd " ++ projectDirName,
+            styleCode $ "    wasp db migrate-dev",
+            styleCode $ "    wasp start",
+            styleText $ "",
+            styleText $ "Check the README for additional guidance!"
+          ]
+    )
+
+{- Functions for styling instructions. They are on purpose of same length, for nicer formatting. -}
+
+styleCode :: String -> String
+styleCode = Term.applyStyles [Term.Bold]
+
+styleText :: String -> String
+styleText = id
+
+{- -}
+
+embeddingsStarterTemplate :: StarterTemplate
+embeddingsStarterTemplate =
+  simpleGhRepoTemplate
+    ("starters", [reldir|embeddings|])
+    ( "embeddings",
+      "Comes with code for generating vector embeddings and performing vector similarity search."
+    )
+    ( \projectDirName ->
+        "To learn how to set up and run your new app, check instructions in "
+          <> projectDirName FP.</> "README.md"
+    )
+
+simpleGhRepoTemplate :: (String, Path' Rel' Dir') -> (String, String) -> StartingInstructionsBuilder -> StarterTemplate
+simpleGhRepoTemplate (repoName, tmplPathInRepo) (tmplDisplayName, tmplDescription) buildStartingInstructions =
+  GhRepoStarterTemplate
+    ( GhRepo.GithubRepoRef
+        { GhRepo._repoOwner = waspGhOrgName,
+          GhRepo._repoName = repoName,
+          GhRepo._repoReferenceName = waspVersionTemplateGitTag
+        }
+    )
+    ( DirBasedTemplateMetadata
+        { _name = tmplDisplayName,
+          _description = tmplDescription,
+          _path = tmplPathInRepo,
+          _buildStartingInstructions = buildStartingInstructions
+        }
+    )
+
+waspGhOrgName :: String
+waspGhOrgName = "wasp-lang"
+
+-- NOTE: As version of Wasp CLI changes, so we should update this tag name here,
+--   and also create it on gh repos of templates.
+--   By tagging templates for each version of Wasp CLI, we ensure that each release of
+--   Wasp CLI uses correct version of templates, that work with it.
+waspVersionTemplateGitTag :: String
```

Comment:
If the tag is not present on the repo, how would that look like for the user in the CLI?

Reply:
It would return HTTP error where it failed to fetch from github URL that contains that tag. I don't know if it is worth spending time on improving that error message -> it shouldn't happen anyway.